apps/wpcom-block-editor/src/wpcom/features/redirect-onboarding-user-after-publishing-post.js

```diff
@@ -0,0 +1,32 @@
+import { select, subscribe } from '@wordpress/data';
+import domReady from '@wordpress/dom-ready';
+import { getQueryArg } from '@wordpress/url';
+
+const siteOrigin = getQueryArg( window.location.search, 'origin' );
+const showLaunchpad = getQueryArg( window.location.search, 'showLaunchpad' );
+
+export const postWasPublished = async () =>
+	new Promise( ( resolve ) => {
+		const unsubscribe = subscribe( () => {
+			const isCurrentPostPublished = select( 'core/editor' ).isCurrentPostPublished();
+			const getCurrentPostRevisionsCount = select( 'core/editor' ).getCurrentPostRevisionsCount();
+
+			if ( isCurrentPostPublished && getCurrentPostRevisionsCount === 1 ) {
+				unsubscribe();
+				resolve();
+			}
+		} );
+	} );
+
+function redirectOnboardingUserAfterPublishingPost() {
+	if ( 'true' !== showLaunchpad ) {
+		return false;
+	}
+
+	postWasPublished().then( () => {
+		const siteSlug = window.location.hostname;
+		window.location.href = siteOrigin + '/setup/write/launchpad?siteSlug=' + siteSlug;
```

Comment:
I left a comment regarding the origin in the thread: p1681812706905759/1681767411.027679-slack-CKZHG0QCR.